### PR TITLE
DEV-2948: fix error saving donation page with empty sidebar_elements

### PR DESCRIPTION
**Pull request by leo-dcfa**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

N/A

#### What's this PR do?

- Updated `sidebar_elements` in `DonationPage` model to accept blank values since the default empty list `[]` is considered blank
- Added corresponding migration
- Updated .pre-commit config for isort - there were some failures trying to push code 

#### Why are we doing this? How does it help us?

To prevent errors when trying to edit a page with no sidebar elements in Django Admin 

#### How should this be manually tested? Please include detailed step-by-step instructions.

- Go to `Django Admin`
- Go to a Donation Page with an empty sidebar element
- Edit the Name of the page
- Hit `Save`
- Assert that the page is saved with a new name

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

N/A

#### Have automated unit tests been added? If not, why?

Yes

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

N/A

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-2948]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

N/A

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

N/A

[DEV-2948]: https://news-revenue-hub.atlassian.net/browse/DEV-2948?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by benjaminEwhite — COMMENTED** [review]
Overall looks good. I left a suggestion about how to shorten the test (and make it more in line with the pytest/fixture-centric approach I'm trying to standardize throughout the app).

Will note in passing that we're not testing the sidebar element as it relates to `Template` model, but that seems fine since `Template` is being retired in DEV-3164 anyway.

File: apps/pages/tests/test_admin.py [hunk 1]
```diff
@@ -204,3 +204,38 @@ def test_make_page_allows_duplicate_name(self):
...
+            "_save",
+        ],
+        "",
+    )
+    data["revenue_program"] = page_empty_sidebar_elements.revenue_program.id
+    data["sidebar_elements"] = "[]"
+    data["name"] = new_name
+    admin_client.post(reverse("admin:pages_donationpage_change", args=[page_empty_sidebar_elements.pk]), data=data)
+    page_empty_sidebar_elements.refresh_from_db()
+    assert page_empty_sidebar_elements.name == new_name
```
Comment (benjaminEwhite) [inline apps/pages/tests/test_admin.py:241]:
```suggestion
@pytest.mark.django_db
def test_can_modify_donation_page_when_sidebar_elements_is_empty(admin_client):
    page_empty_sidebar_elements = DonationPageFactory(sidebar_elements=[])
    admin_client.post(
        reverse("admin:pages_donationpage_change", args=[page_empty_sidebar_elements.pk]),
        data=dict.fromkeys(
            [
                "csrfmiddlewaretoken",
                "published_date_0",
                "published_date_1",
                "slug",
                "name",
                "thank_you_redirect",
                "post_thank_you_redirect",
                "header_bg_image",
                "header_logo",
                "header_link",
                "heading",
                "graphic",
                "styles",
                "elements",
                "initial-elements",
                "sidebar_elements",
                "initial-sidebar_elements",
                "_save",
            ],
            "",
        )
        | {
            "revenue_program": page_empty_sidebar_elements.revenue_program.id,
            "sidebar_elements": "[]",
            "name": (new_name := "New name"),
        },
    )
    page_empty_sidebar_elements.refresh_from_db()
    assert page_empty_sidebar_elements.name == new_name
```

**Review by benjaminEwhite — APPROVED** [review]
Looks good. Have one suggestion for making test more terse, but you can take that or leave it .

**Review by x110dc — APPROVED** [review]

**Comment by benjaminEwhite** [comment]
Closing, tearing down, then reopening as part of work for [DEV-3243 | fix hookdeck webhooks](https://news-revenue-hub.atlassian.net/browse/DEV-3243), to trigger fresh review app.

**Pull request merged**